crates/app/cli.rs

```diff
@@ -1,27 +1,210 @@
+    )]
+    image_path: Option<String>,
+
+    #[arg(
+        short = 'R',
+        long,
+        help = "Resident",
+        long_help = "Prevents automatic removal of notifications after an action"
+    )]
+    resident: Option<bool>,
+
+    #[arg(
+        short,
+        long,
+        help = "Sound file",
+        long_help = "Path to a sound file to play when the notification pops up"
+    )]
+    sound_file: Option<String>,
+
+    #[arg(
+        short = 'N',
+        long,
+        help = "Sound name",
+        long_help = "A themeable sound name to play when the notification pops up"
+    )]
+    sound_name: Option<String>,
+
+    #[arg(
+        short = 'S',
+        long,
+        help = "Suppress sound",
+        long_help = "Causes the server to suppress playing any sounds"
+    )]
+    suppress_sound: Option<bool>,
+
+    #[arg(
+        short = 'T',
+        long,
+        help = "Transient",
+        long_help = "Marks the notification as transient, bypassing the server's persistence capability if available"
+    )]
+    transient: Option<bool>,
+
+    #[arg(
+        short = 'C',
+        long,
+        help = "Action icons",
+        long_help = "Interprets action IDs as icons, annotated by display names"
+    )]
+    action_icons: Option<bool>,
 }
 
 impl Args {
     pub async fn process(&self) -> anyhow::Result<()> {
-        match self {
-            Args::Run => {
-                let _ = &*CONFIG; // Initializes the configuration.
+        if let Args::Run = self {
+            self.run().await?
+        }
 
-                backend::run().await?;
-            }
+        let noti = client::NotiClient::init().await?;
+
+        match self {
+            Args::Run => unreachable!(),
+            Args::Send(args) => self.send(noti, args).await?,
+            Args::ServerInfo => self.server_info(noti).await?,
         }
 
         Ok(())
     }
+
+    async fn run(&self) -> anyhow::Result<()> {
+        let _ = &*CONFIG; // Initializes the configuration.
+
+        backend::run().await
+    }
+
+    async fn send(&self, noti: client::NotiClient<'_>, args: &SendCommand) -> anyhow::Result<()> {
+        let hints_data = client::HintsData {
+            urgency: args.urgency.clone(),
+            category: args.category.clone(),
+            desktop_entry: args.desktop_entry.clone(),
+            image_path: args.image_path.clone(),
+            sound_file: args.sound_file.clone(),
+            sound_name: args.sound_name.clone(),
+            resident: args.resident,
+            suppress_sound: args.suppress_sound,
+            transient: args.transient,
+            action_icons: args.action_icons,
+        };
+
+        noti.send_notification(
+            args.replaces_id,
+            &args.app_name,
+            &args.icon,
+            &args.summary,
+            &args.body,
+            args.timeout,
+            &args.actions,
+            &args.hints,
+            &hints_data,
+        )
```

Comment:
According to my words in previous comment, maybe here you also can remove references. If you can't - leave it.